src/main/scala/is/hail/expr/ir/Pretty.scala

```diff
@@ -184,11 +184,11 @@ object Pretty {
               prettyBooleanLiteral(dropRows) + " " +
               '"' + StringEscapeUtils.escapeString(Serialization.write(reader)(MatrixReader.formats)) + '"'
             case TableToMatrixTable(_, rowKey, colKey, rowFields, colFields, partitionKey, nPartitions) =>
-              prettyStrings(rowKey)
-              prettyStrings(colKey)
-              prettyStrings(rowFields)
-              prettyStrings(colFields)
-              prettyStrings(partitionKey)
+              prettyStrings(rowKey) + " " +
```

Comment:
oops.

Reply:
why didn't this fail tests? I thought it was tested...

Reply:
Nobody has added Matrix parse tests to Python yet.

Reply:
I thought this was getting tested in Scala

Reply:
I may have neglected to add it to matrixIRs() in IRSuite

Reply:
You did, but the data provider throw an error constructing the TableToMatrixTable and it was silently ignored.  I fixed that (see my changes to IRSuite), but we should figure out how to make that fail the test somehow.

Reply:
This should have been fixed in 2017 https://github.com/cbeust/testng/issues/217. Do we have a compelling reason to use a version of TestNG from February 2015? I suggest we jump to 6.14.3. https://mvnrepository.com/artifact/org.testng/testng

Reply:
ping @jigold who was looking at this

Reply:
Sorry, my comment was below. I tried the newest version of testng and it didn't fix the problem. See my PR #3964 for a fix.

Reply:
Meaning the data provider threw an exception but did not fail the test? That PR in testng should have addressed exactly this issue.